features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/telemetry/internal/TelemetryEventHandler.kt

```diff
@@ -33,15 +36,20 @@ internal class TelemetryEventHandler(
     internal val eventSampler: Sampler,
     internal val configurationExtraSampler: Sampler =
         RateBasedSampler(DEFAULT_CONFIGURATION_SAMPLE_RATE),
-    private val maxEventCountPerSession: Int = MAX_EVENTS_PER_SESSION
+    private val maxEventCountPerSession: Int = MAX_EVENTS_PER_SESSION,
+    internalSdkCore: InternalSdkCore? = Datadog.getInstance() as? InternalSdkCore,
+    private val deviceInfo: DeviceInfo? = internalSdkCore?.getDatadogContext()?.deviceInfo
```

Comment:
why this is needed, if inside `handleEvent` method there is already `DatadogContext` provided as a part of `withWriteContext` lambda?